### ISSUE TITLE
Wire up stats and dashboards for DaemonSets

### ISSUE THREAD

**Issue by klingerf** [linkerd/linkerd2#2006]
Similar to #1983, we should update the CLI, web UI, and grafana dashboards to start displaying stats for daemonsets. The `daemonset` label already exists in Prometheus, so this should not require any proxy changes.

**Comment by zknill** [comment]
Hey; this is currently tagged as `good first issue` and `help wanted`; are PRs for this welcome? I am happy to have a crack at it!

**Comment by grampelberg** [comment]
@zknill Yeah!

**Comment by wmorgan** [comment]
@zknill do it!

**Comment by zknill** [comment]
I've made some progress on this; and almost ready to open a PR for the controller and cli.

There are a few things that are unclear to me about the web UI;
- Currently graphs between resources are rendered considering only resources of the same type. E.g. if everything from the example app were a `DaemonSet` then the graph could look like;

![image](https://user-images.githubusercontent.com/6664336/51129796-4c079c80-1823-11e9-8735-01f0e8941ddf.png)

Is this what's required? It feels to me that this is not immediately useful; and there might be some need for a cross over between resources e.g. deployments vs daemonsets etc.

Secondly; the dependency graph on the namespace page works only with deployments and it's not immediately obvious to me how to change that. If this is even something that should happen?

![image](https://user-images.githubusercontent.com/6664336/51129917-a43e9e80-1823-11e9-88ae-5e71d47894b5.png)


TL;DR the question is;
- Is it fine for information to only be displayed in graphical format for the resource that match the type of the current resource. deployments vs. deployments, daemonsets vs. daemonsets etc?

**Comment by dadjeibaah** [comment]
@zknill Looks like you are making great progress! Thanks for picking up this issue. 

> Is it fine for information to only be displayed in a graphical format for the resource that match the type of the current resource. deployments vs. deployments, daemonsets vs. daemonsets etc?

I do agree that it might be useful to have a more expanded view of the current resource's upstream and downstream's services. However, I think this is a larger issue we would need to address in a separate PR. I think there is an [issue](https://github.com/linkerd/linkerd2/issues/911) that kind of relates to this and we would need to fix this (possibly other issues) in order to get that kind of info on the dashboard. 

The good thing is that we still get inbound stats for that particular resource even though, graphically, we are unable to view downstream or upstream resources. e.g. A replication controller resource traffic from a deployment. 

<img width="1208" alt="screen shot 2019-01-14 at 11 24 22 am" src="https://user-images.githubusercontent.com/2197104/51135626-60ca3d00-17ef-11e9-88ca-bc142f44fabb.png">

So I think scoping this down to just having the dashboard display what you have now is a good start, then we would reopen/create any PRs that would help us expand the view of upstream and downstream resources.

The same goes for the grafana dashboard as well.

**Comment by zknill** [comment]
Excellent; this is the answer I expected (and hoped for)! Thanks for the speedy response.

**Comment by dadjeibaah** [comment]
Closed by PR #2086
